### PR TITLE
Update managed-identity.md

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 1]
```diff
@@ -18,6 +18,15 @@ A common challenge for developers is the management of secrets, credentials, cer
...
 For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
 
+## Which SDK to use - Azure SDK or MSAL?
+
+Both MSAL.NET and Azure SDK allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher level API via its [DefaultAzureCredential](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme?view=azure-dotnet#defaultazurecredential) abstraction. 
+
+If your application already uses one of the SDKs, continue using the same SDK. If you are writing a new application, use Azure SDK if you plan to call other Azure resources, as the SDK integrates better. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web api. 
+
+>[!Note] 
+>[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher level API that offers integration with ASP.NET Core and ASP.NET Classic. Internally it uses MSAL.
```
Comment (bgavrilMS) [inline msal-dotnet-articles/advanced/managed-identity.md:28]: CC @jmprieur to update this, I am not sure how to use ID.Web with MI

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 2]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
 
 For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
 
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 / OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher level API via its DefaultAzureCredential and ManagedIdentityCredential abstractions.
```
Comment (localden) [inline msal-dotnet-articles/advanced/managed-identity.md:25]:
```suggestion
Both MSAL.NET and [Azure SDK](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher-level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions. 
```

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 3]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
 
 For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
 
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 / OIDC protocols.
```
Comment (localden) [inline msal-dotnet-articles/advanced/managed-identity.md:23]:
```suggestion
MSAL libraries provide lower level APIs that are closer to the OAuth2 and OIDC protocols. 
```

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 4]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
...
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 / OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher level API via its DefaultAzureCredential and ManagedIdentityCredential abstractions. 
+
+If your application already uses one of the SDKs, continue using the same SDK. If you are writing a new application, use Azure SDK if you plan to call other Azure resources, as the SDK integrates better and provides a good developer experience by allowing the app to be ran on developer private machines where Managed Identity doesn't exist. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web api. 
+
+>[!Note] 
+>[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher level API that offers integration with ASP.NET Core and ASP.NET Classic. It supports Managed Identity as well. Internally it uses MSAL.
```
Comment (localden) [inline msal-dotnet-articles/advanced/managed-identity.md:30]:
```suggestion
[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher-level API that offers integration with ASP.NET Core and ASP.NET Classic, while using MSAL under the hood. The library also supports Managed Identity.
```

**Review by localden — CHANGES_REQUESTED** [review]

File: msal-dotnet-articles/advanced/managed-identity.md (hunk 2 above)
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:25]:
```suggestion
Both MSAL.NET and [Azure SDK](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via managed identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions. 
```

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 5]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
 
 For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
 
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 / OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](https://learn.microsoft.com/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher level API via its DefaultAzureCredential and ManagedIdentityCredential abstractions. 
+
+If your application already uses one of the SDKs, continue using the same SDK. If you are writing a new application, use Azure SDK if you plan to call other Azure resources, as the SDK integrates better and provides a good developer experience by allowing the app to be ran on developer private machines where Managed Identity doesn't exist. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web api.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:27]:
```suggestion
If your application already uses one of the SDKs, continue using the same SDK. Use Azure SDK, if you are writing a new application and plan to call other Azure resources, as this SDK provides a better developer experience by allowing the app to run on private developer machines where managed identity doesn't exist. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web API. 
```

**Review by pmaytak — APPROVED** [review]

**Comment by localden** [comment]
@pmaytak could you please update your suggestions based on latest commits?

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 6]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
 
 For a complete list, refer to [Azure services that can use managed identities to access other services](/azure/active-directory/managed-identities-azure-resources/managed-identities-status).
 
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 and OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher-level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:25]:
```suggestion
Both MSAL.NET and [Azure SDK](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via managed identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher-level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions. 
```

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 7]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
...
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 and OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via Managed Identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher-level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions. 
+
+If your application already uses one of the SDKs, continue using the same SDK. Use Azure SDK, if you are writing a new application and plan to call other Azure resources, as this SDK provides a better developer experience by allowing the app to run on private developer machines where managed identity doesn't exist. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web API. 
+
+>[!Note] 
+[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher-level API that offers integration with ASP.NET Core and ASP.NET Classic, while using MSAL under the hood. The library also supports Managed Identity.
```
Comment (pmaytak) [inline msal-dotnet-articles/advanced/managed-identity.md:30]:
```suggestion
[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher-level API that offers integration with ASP.NET Core and ASP.NET Classic, while using MSAL under the hood. The library also supports token acquisition with managed identity.
```

**Comment by localden** [comment]
@jmprieur is the `Microsoft.Identity.Web.ITokenAcquisition.ReplyForbiddenWithWwwAuthenticateHeader` entity really gone from the package?

**Review by localden — APPROVED** [review]

**Review by localden — APPROVED** [review]

**Review by localden — DISMISSED** [review]

**Comment by localden** [comment]
@bgavrilMS updated. Should be going into `main` now.

**Review by localden — APPROVED** [review]

File: msal-dotnet-articles/advanced/managed-identity.md [hunk 8]
```diff
@@ -18,6 +18,17 @@ A common challenge for developers is the management of secrets, credentials, cer
...
+## Which SDK to use - Azure SDK or MSAL?
+
+MSAL libraries provide lower level APIs that are closer to the OAuth2 and OIDC protocols. 
+
+Both MSAL.NET and [Azure SDK](/dotnet/api/overview/azure/identity-readme?view=azure-dotnet&preserve-view=true) allow to acquire tokens via managed identity. Internally, Azure SDK uses MSAL.NET, and it provides a higher-level API via its `DefaultAzureCredential` and `ManagedIdentityCredential` abstractions. 
+
+If your application already uses one of the SDKs, continue using the same SDK. Use Azure SDK, if you are writing a new application and plan to call other Azure resources, as this SDK provides a better developer experience by allowing the app to run on private developer machines where managed identity doesn't exist. Consider using MSAL if you need to call other downstream web APIs like Microsoft Graph or your own web API. 
+
+>[!Note] 
+[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher-level API that offers integration with ASP.NET Core and ASP.NET Classic, while using MSAL under the hood. The library also supports token acquisition with managed identity.
```
Comment (jmprieur) [inline msal-dotnet-articles/advanced/managed-identity.md:30]:
```suggestion
[Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web) is a higher-level API that offers integration with ASP.NET Core and ASP.NET Classic, while using MSAL under the hood. The library also provides a way to load credentials (certificates, signed assertions) used by MSAL.NET as client credentials. For certificates it uses the `DefaultAzureCredentials` to fetch certificates from KeyVault. It also offers workload identity federation with managed identity credentials. For details see [CredentialDescription](/dotnet/api/microsoft.identity.abstractions.credentialdescription.keyvaulturl?view=msal-model-dotnet-latest#microsoft-identity-abstractions-credentialdescription-keyvaulturl&preserve-view=true)
```

**Comment by jmprieur** [comment]
> @jmprieur is the `Microsoft.Identity.Web.ITokenAcquisition.ReplyForbiddenWithWwwAuthenticateHeader` entity really gone from the package?

It should still be there, @localden, but in the Ms.Id.Web.TokenAcquisition NuGet package (since v2.0)

**Pull request merged**